src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -387,7 +387,7 @@ public void Initialize(ISystemClock? clock = null)
                 ResponseGrpcEncoding = GrpcProtocolConstants.IdentityGrpcEncoding;
             }
 
-            HttpContext.Response.Headers.Append(GrpcProtocolConstants.MessageEncodingHeader, ResponseGrpcEncoding);
+            HttpContext.Response.Headers[GrpcProtocolConstants.MessageEncodingHeader] = ResponseGrpcEncoding;
```

Comment:
This has a functional difference right? One is adding to existing headers, one is overwriting? Is that what we want?

Reply:
Overwriting is correct. It is also slightly faster.